plasoscaffolder/plugins/manager.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+"""The plugin manager."""
+
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Generator
+from typing import Tuple
+from typing import Type
+
+from plasoscaffolder.plugins import interface
+
+
+class PluginManager(object):
+  """The plugin manager."""
+
+  _plugin_classes = {}
+
+  @classmethod
+  def DeregisterPlugin(cls, plugin_class: Type[interface.ScaffolderPlugin]):
+    """Deregisters a plugin class.
+
+    The plugin classes are identified based on their lower case name.
+
+    Args:
+      plugin_class (type): plugin class (subclass of ScaffolderPlugin).
+
+    Raises:
+      KeyError: if plugin class is not set for the corresponding name.
+    """
+    plugin_provides = plugin_class.PROVIDES.lower()
+    if plugin_provides not in cls._plugin_classes:
+      raise KeyError('Plugin class not set for name: {0:s}.'.format(
+          plugin_class.PROVIDES))
+
+    del cls._plugin_classes[plugin_provides]
+
+  @classmethod
+  def GetPluginNames(cls) -> Generator[str, None, None]:
+    """Retrieves the plugin names.
+
+    Yields:
+      str: plugin names.
+    """
+    for plugin_provides, plugin_class in cls.GetPlugins():
+      yield plugin_provides
+
+  @classmethod
+  def GetPluginInformation(cls) -> Generator[Tuple[str, str], None, None]:
+    """Retrieves the plugin information.
+
+    Yields:
+      tuple[str, str]: pairs of plugin names and descriptions.
+    """
+    for plugin_provides, plugin_class in cls.GetPlugins():
+      description = getattr(plugin_class, 'DESCRIPTION', '')
```

Comment:
DESCRIPTION is already initialized to the empty string, so just do plugin_class.DESCRIPTION